pkg/apis/core/validation/validation.go

```diff
@@ -2183,7 +2190,11 @@ func validateEnvVarValueFrom(ev core.EnvVar, fldPath *field.Path) field.ErrorLis
 	}
 	if ev.ValueFrom.ResourceFieldRef != nil {
 		numSources++
-		allErrs = append(allErrs, validateContainerResourceFieldSelector(ev.ValueFrom.ResourceFieldRef, &validContainerResourceFieldPathExpressions, fldPath.Child("resourceFieldRef"), false)...)
+		localValidContainerResourceFieldPathPrefixes := sets.NewString()
+		if utilfeature.DefaultFeatureGate.Enabled(features.DownwardAPIHugePages) {
```

Comment:
same comment about needing the old data to allow no-op updates

Reply:
see: https://github.com/kubernetes/kubernetes/pull/86102#discussion_r518312814